microsoft/fhir-server
# Add Tests for Jobs and Client

## Description
While testing importOrchestratorJob, I discovered several issues in job tests. Essentially the issues are as following:

- Need more tests for OrchestratorJob such as testing different job status combinations.
- The function GetJobByGroupId in TestQueueClient only read jobInfo from local variable.
- Missing test when job was cancelled by user and job was cancelled by itself at the same time, it is better to verify the job status in this situation.

This PR will add several tests for OrchestratorJob and SqlQueueClient, and upgrade the function for TestQueueClient.

## Testing

- Added some tests for OrchestratorJob:
  - Verify the thrown exception when some jobs cancelled or failed.
  - Verify the thrown exception and job status when job was cancelled or failed after several calls.
  - Verify the job status when job was cancelled or failed and other jobs were running or created.
 - Upgraded TestQueueClient, added func for GetJobByGroupId.
 - Added test for SqlQueueClientTest to verify job status after cancelJobByGroupId called twice.

## FHIR Team Checklist
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)
